### PR TITLE
feat!: Persist learning_rate, power_t, and initial_t for loaded models

### PR DESCRIPTION
**Pull request by bassmang**
Addresses: #3179

### CODE REVIEW

File: vowpalwabbit/parse_args.cc [hunk 1]
```diff
@@ -1381,7 +1392,6 @@ VW::workspace& parse_args(
 
     parse_diagnostics(*all.options, all);
 
-    all.initial_t = static_cast<float>(all.sd->t);
```
Comment (JohnLangford) [inline vowpalwabbit/parse_args.cc:1384]: Why this change?

Comment (bassmang) [reply vowpalwabbit/parse_args.cc:1384]: This line has been moved to `1093` in the same file. The entire option_group_definition had to be moved to a later section in the parser, namely after
`load_header_merge_options(*all.options, all, *model, interactions_settings_duplicated);`
is called so that we have access to the loaded arguments. This line needs to be moved because it is set of the parsed arguments.

Comment (JohnLangford) [reply vowpalwabbit/parse_args.cc:1384]: Sounds good.

File: vowpalwabbit/parse_args.cc [hunk 2]
```diff
@@ -1070,6 +1070,29 @@ void parse_example_tweaks(options_i& options, VW::workspace& all)
...
+               .default_value(0.5f)
+               .keep()
+               .allow_override()
+               .help("Set learning rate")
+               .short_name("l"))
+      .add(make_option("power_t", all.power_t).default_value(0.5f).keep().allow_override().help("T power value"))
+      .add(make_option("decay_learning_rate", all.eta_decay_rate)
+               .default_value(1.f)
+               .help("Set Decay factor for learning_rate between passes"))
+      .add(make_option("initial_t", all.sd->t).keep().allow_override().help("Initial t value"))
```
Comment (jackgerrits) [inline vowpalwabbit/parse_args.cc:1087]: Why is this needed as keep. Both t and initial_t are serialized as part of a save_resume model

Comment (bassmang) [reply vowpalwabbit/parse_args.cc:1087]: I dont think being part of save_resume is sufficient here. If you save and load a model without `keep()`, the output will print `initial_t` with the default value (although internally initial_t and sd->t should restore properly)

Comment (bassmang) [reply vowpalwabbit/parse_args.cc:1087]: save_resume isn't sufficient here because it is called after the initial output printing, so if you save and load a model with some `initial_t` it will print the default value (although internally it should restore initial_t at a later point)

Comment (jackgerrits) [reply vowpalwabbit/parse_args.cc:1087]: But now dont we have duplicated/potentially conflicting initial_t information between the command line and the model?

Comment (bassmang) [reply vowpalwabbit/parse_args.cc]: I removed `keep()` and `allow_override()` from `initial_t`, and moved the printing to after the model is loaded. This is actually a bug fix since `initial_t` is often altered before being save_loaded, and was previously printing out the default for loaded models.

**Review by jackgerrits — APPROVED** [review]
